src/types.ts

```diff
@@ -25,3 +25,25 @@ export enum SelectionType {
   Hide,
   Highlight,
 }
+
+export enum AgentType {
+  Agent,
+  DisplayState,
+}
+
+interface BaseCustomCheckboxProps {
+  agentName: string;
+  agentType: AgentType;
+}
+
+export interface AgentProps extends BaseCustomCheckboxProps {
+  agentType: AgentType.Agent;
+  displayStateName?: never;
```

Comment:
why can't this be a string?

Reply:
The two enum members? It could be. I’ve been given the feedback when using strings in this way to formalize them and use an enum..